### PR TITLE
Updating for 1.8.9 - Spells

### PR DESCRIPTION
**Pull request by MageProtocol**

### CODE REVIEW

File: src/main/java/am2/codechicken/LightningBoltCommon.java [hunk 1]
```diff
@@ -168,7 +169,8 @@ private float rayTraceResistance(AMVector3 start, AMVector3 end, float prevresis
 			return prevresistance;
 		}
 		if (mop.typeOfHit == MovingObjectType.BLOCK){
-			Block block = this.world.getBlock(mop.blockX, mop.blockY, mop.blockZ);
+			BlockPos pos = new BlockPos(mop.getBlockPos().getX(), mop.getBlockPos().getY(), mop.getBlockPos().getZ());
+			Block block = this.world.getBlockState(pos).getBlock();
```
Comment (williewillus) [inline src/main/java/am2/codechicken/LightningBoltCommon.java]: how about just world.getBlockState(mop.getBlockPos()).getBlock() :P

File: src/main/java/am2/power/PowerNodeRegistry.java [hunk 2]
```diff
@@ -459,16 +462,18 @@ public int compare(IPowerNode o1, IPowerNode o2){
 			TileEntity te1 = (TileEntity)o1;
 			TileEntity te2 = (TileEntity)o2;
 
-			if (te1.xCoord == te2.xCoord && te1.zCoord == te2.zCoord && te1.yCoord == te2.yCoord)
+			BlockPos tePos1 = te1.getPos();
+			BlockPos tePos2 = te2.getPos();
+			if (tePos1.getX() == tePos2.getX() && tePos1.getZ() == tePos2.getZ() && tePos1.getY() == tePos2.getY())
```
Comment (williewillus) [inline src/main/java/am2/power/PowerNodeRegistry.java]: te1.getPos().equals(te2.getPos())

File: src/main/java/am2/codechicken/LightningBoltCommon.java (hunk 1 above)
Comment (Kolatra) [inline src/main/java/am2/codechicken/LightningBoltCommon.java]: Yeah that'll be a lot cleaner.

File: src/main/java/am2/spell/components/Drought.java [hunk 3]
```diff
@@ -22,51 +24,49 @@
...
-			world.setBlock(blockx, blocky, blockz, Blocks.sand);
+			world.setBlockState(pos, Blocks.sand.getDefaultState());
 			return true;
 		}else if (block == Blocks.stone){
-			world.setBlock(blockx, blocky, blockz, Blocks.cobblestone);
+			world.setBlockState(pos, Blocks.cobblestone.getDefaultState());
 			return true;
-		}else if (block == Blocks.stonebrick && world.getBlockMetadata(blockx, blocky, blockz) != 2){
+		}else if (block == Blocks.stonebrick && world.getBlockState(pos).getBlock().getMetaFromState(world.getBlockState(pos)) != 2){
 			world.setBlockMetadataWithNotify(blockx, blocky, blockz, 2, 2);
```
Comment (williewillus) [inline src/main/java/am2/spell/components/Drought.java]: check blockstates properly

File: src/main/java/am2/spell/components/Drought.java [hunk 4]
```diff
@@ -22,51 +24,49 @@
...
+				case WEST:
+					pos = pos.west();
+					break;
+				case DOWN:
+					pos = pos.down();
+					break;
+				case UP:
+					pos = pos.up();
+					break;
 			}
```
Comment (williewillus) [inline src/main/java/am2/spell/components/Drought.java]: whole switch can be collapsed into pos = pos.offset(EnumFacing)

File: src/main/java/am2/spell/components/Ignition.java [hunk 5]
```diff
@@ -22,33 +24,32 @@
...
+				break;
+			case NORTH:
+				pos = pos.north();
+				break;
+			case SOUTH:
+				pos = pos.south();
+				break;
+			case WEST:
+				pos = pos.west();
+				break;
```
Comment (williewillus) [inline src/main/java/am2/spell/components/Ignition.java]: same here pos = pos.offset(facing)

File: src/main/java/am2/spell/components/Light.java [hunk 6]
```diff
@@ -49,31 +75,31 @@ public boolean applyEffectBlock(ItemStack stack, World world, int blockx, int bl
...
+					break;
+				case SOUTH:
+					pos = pos.south();
+					break;
+				case EAST:
+					pos = pos.east();
+					break;
+				case WEST:
+					pos = pos.west();
+					break;
```
Comment (williewillus) [inline src/main/java/am2/spell/components/Light.java]: same again

File: src/main/java/am2/spell/components/PlaceBlock.java [hunk 7]
```diff
@@ -90,31 +91,31 @@ public boolean applyEffectBlock(ItemStack stack, World world, int blockx, int bl
...
+					case SOUTH:
+						pos = pos.south();
+						break;
+					case WEST:
+						pos = pos.west();
+						break;
+					case EAST:
+						pos = pos.east();
+						break;
 				}
```
Comment (williewillus) [inline src/main/java/am2/spell/components/PlaceBlock.java]: same :P

File: src/main/java/am2/spell/components/Rift.java [hunk 8]
```diff
@@ -52,18 +54,18 @@ public boolean applyEffectBlock(ItemStack stack, World world, int blockx, int bl
...
+		if (facing == EnumFacing.UP){
+			storage.setPosition(pos.getX() + 0.5, pos.getY() + 1.5, pos.getZ()+ 0.5);
+		}else if (facing == EnumFacing.NORTH){
+			storage.setPosition(pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() - 1.5);
+		}else if (facing == EnumFacing.SOUTH){
+			storage.setPosition(pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 1.5);
+		}else if (facing == EnumFacing.WEST){
+			storage.setPosition(pos.getX() - 1.5, pos.getY() + 0.5, pos.getZ() + 0.5);
+		}else if (facing == EnumFacing.EAST){
+			storage.setPosition(pos.getX() + 1.5, pos.getY() + 0.5, pos.getZ() + 0.5);
```
Comment (williewillus) [inline src/main/java/am2/spell/components/Rift.java]: storage.setPosition(pos.offset(facing).add(0.5, 0.5, 0.5))

Comment (Kolatra) [inline src/main/java/am2/spell/components/Rift.java]: That one actually requires three arguments it looks like.

Comment (williewillus) [inline src/main/java/am2/spell/components/Rift.java]: ah it's a decimal, my bad

**Comment by Kolatra** [comment]
What I have decided for 1.8.9 PRs due to a few reasons is to not merge any of them, however I will include the changed code myself and give appropriate credit for code. Don't let this discourage you, you're doing great work.

**Pull request closed without merging**